src/content/tools/pub/dependencies.md

```diff
@@ -474,6 +474,34 @@ If you modify the pubspec and update to a new major version,
 then you might encounter breaking changes,
 so you need to test even more thoroughly.
 
+### Test with downgraded dependencies
+
+When developing packages for publication, it is often preferable to allow as a wide dependency constraints as possible.
+A wide dependency constraint reduces the likelihood that package consumers face a version resolution conflict.
+
+Hence, if you have a dependency on `foo: ^1.2.3` and version `1.3.0` of `foo` is released, it may be reasonable to keep the existing dependency constraint (`^1.2.3`).
+But if your package starts using features that were added in `1.3.0`, then you'll need to bump your constraint to `^1.3.0`.
+
+However, it's easy to forget to bump a dependency constraint when it becomes necessary.
+Thus, it's best practice to test you package against downgraded dependencies before publishing.
+
+Test against downgraded dependencies by running:
+
+ * [`dart pub downgrade`][],
+ * `dart analyze`, and possibly,
+ * `dart test`.
+
+Testing with downgraded dependencies should happen alongside normals tests with latest dependencies.
```

Comment:
Should we also suggest running `pub downgrade` on the oldest SDK you support, because that is the most likely to give you the oldest version solves?

Reply:
We have no plans to do this in `pana` yet, would be interesting to consider.

I think it's an okay suggestion to make. But will leave it out of this PR, because I want it to land, so that we can point pana error message at it :D

Also the Dart SDK has `@Since('3.0.0')` annotations and there are diagnostics to help you bump the SDK constraint, if it's too low. These annotations and diagnostics aren't perfect, but they are much better than what we have in package land -- where there is nothing telling to bump your constraint when you use a newly added class/method/function/...

Reply:
My main concern was that if people run `pub downgrade` on a new SDK, they might not actually get the older versions of dependencies which they claim to support (that don't support that newer SDK).

I am less concerned about the actual SDK constraint being tested, because as you say `@Since` etc help a lot with that (when used...).